src/bedrock/hash_ring/hash_ring.cpp

```diff
@@ -16,7 +16,7 @@
 
 // assuming the replication factor will never be greater than the number of nodes in a tier
 // return a set of ServerThread that are responsible for a key
-unordered_set<ServerThread, ThreadHash> responsible_global(string key, unsigned global_rep, GlobalHashRing& global_hash_ring) {
+unordered_set<ServerThread, ThreadHash> responsible_global(const string& key, unsigned global_rep, GlobalHashRing& global_hash_ring) {
```

Comment:
This is more of a question than a piece of feedback: What is the performance implication of passing a `const` reference (and a regular reference) rather than passing by value?